comp/otelcol/ddflareextension/impl/configstore.go

```diff
@@ -15,17 +16,41 @@ import (
 )
 
 type configStore struct {
-	provided *otelcol.Config
-	enhanced *otelcol.Config
-	mu       sync.RWMutex
+	provided                *otelcol.Config
+	enhanced                *otelcol.Config
+	mu                      sync.RWMutex
+	providedConfigSupported bool
+	// TODO: Replace enhanced config with effective config once dependencies
```

Comment:
You should be able to remove all this once #30061 is merged and you pull in main 🙂

Reply:
I plan to keep the providedConfigSupported so that we can still use logic to return a "unsupported" or nil value in the provided config field but yes thank you for that, I appreciate it